pkg/cloudprovider/aws/aws.go

```diff
@@ -313,3 +328,96 @@ func (n *NodeGroup) setASGDesiredSize(newSize int64) error {
 	_, err := n.provider.service.SetDesiredCapacity(input)
 	return err
 }
+
+// setASGDesiredSizeOneShot uses the AWS fleet API to acquire all desired
+// capacity in one step and then add it to the existing auto-scaling group.
+func (n *NodeGroup) setASGDesiredSizeOneShot(addCount int64) error {
+	fleet, err := n.provider.ec2_service.CreateFleet(&ec2.CreateFleetInput{
+		Type:                             awsapi.String("instant"),
+		TerminateInstancesWithExpiration: awsapi.Bool(false),
+		OnDemandOptions: &ec2.OnDemandOptionsRequest{
+			MinTargetCapacity:      awsapi.Int64(addCount),
+			SingleAvailabilityZone: awsapi.Bool(true),
+			SingleInstanceType:     awsapi.Bool(true),
+		},
+		TargetCapacitySpecification: &ec2.TargetCapacitySpecificationRequest{
+			OnDemandTargetCapacity:    awsapi.Int64(addCount),
+			TotalTargetCapacity:       awsapi.Int64(addCount),
+			DefaultTargetCapacityType: awsapi.String("on-demand"),
+		},
+		LaunchTemplateConfigs: []*ec2.FleetLaunchTemplateConfigRequest{
+			&ec2.FleetLaunchTemplateConfigRequest{
+				LaunchTemplateSpecification: &ec2.FleetLaunchTemplateSpecificationRequest{
+					LaunchTemplateId: awsapi.String(n.launchTemplateId),
+					Version:          awsapi.String(n.launchTemplateVersion),
+				},
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// This will hold any launch errors for the fleet. In the case of an
+	// instant fleet with a single instant type this will indicate that the
+	// entire fleet failed to launch.
+	for _, lerr := range fleet.Errors {
+		return errors.New(*lerr.ErrorMessage)
+	}
+
+	instances := make([]*string, 0)
+	for _, i := range fleet.Instances {
+		instances = append(instances, i.InstanceIds...)
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	deadline := time.NewTimer(1 * time.Minute)
+	defer ticker.Stop()
+	defer deadline.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			if n.allInstancesReady(instances) {
+				break
+			}
+		case <-deadline.C:
+			return errors.New("Not all instances could be started")
+		}
+	}
+
+	_, err = n.provider.service.AttachInstances(&autoscaling.AttachInstancesInput{
+		AutoScalingGroupName: awsapi.String(n.id),
+		InstanceIds:          instances,
```

Comment:
Looking at the documentation for `AttachInstancesInput` (https://docs.aws.amazon.com/sdk-for-go/api/service/autoscaling/#AttachInstancesInput) it looks like there is a limit of 20 instances and if we exceed this the API call will fail. 

Could you add some logic to handle this?